status/contacts.nim

```diff
@@ -134,8 +135,4 @@ proc contactRequestReceived*(self: ContactModel, id: string): bool =
   contact.systemTags.contains(contactRequest)
 
 proc rejectContactRequest*(self: ContactModel, id: string) =
-  let contact = self.getContactByID(id)
-  contact.systemTags.delete(contact.systemTags.find(contactRequest))
-
-  discard self.saveContact(contact)
-  self.events.emit("contactRemoved", Args())
+  self.removeContact(id)
```

Comment:
What about contact blocked? Can you receive a contactRequest from a blocked contact?

Reply:
@alaibe in order to block any contact it must be firstly in your contact list. So you need to accept it and then block it, what further means that you're still "friend" with that contact but you blocked him, so you won't be able to see his new messages.

Reply:
I think a contact can be in your list without having any system tags. For example when you see ens name of people you are not contact added with. But they are part of your contact list

That means that you can block anyone

Reply:
how that could be possible? Blocking someone should not depend on being in contacts or not. I want to be able to block u and dont see your messages without adding you to contacts.

So specifically `that if I block someone, then remove him from my contact list, I will see the message of this person in public chat.` is not expected by designs made: https://www.figma.com/file/Mr3rqxxgKJ2zMQ06UAKiWL/💬-Chat⎜Desktop?node-id=8376%3A205314

Reply:
Yes I see now that you can block even a person who is not your contact, but in this case that blocked person should not be able to send you a contact request or a message (I guess).

Not sure about this `I think a contact can be in your list without having any system tags` is that a bug or feature?

Not sure that I understand this `So that means that if I block someone, then remove him from my contact list, I will see the message of this person in public chat`

Reply:
@alaibe all in all for that part we need exact requests and revision it, since I find it a bit confusing.

Reply:
I don't know what is the correct feature request, but if you block/remove someone you're not able to have 1:1 chat with him, and he is unable to send you a message, but you should be able to see his messages in any public chat (I understood you that you're saying that I cannot see messages of blocked/removed user in any public chat)?

Reply:
In removeContact there is this line:
```
contact.systemTags = @[]
```

That means that if you remove someone from your friend list (someone you can have 1:1 chat with), you will at the same time remove him from your blocked list, so you will receive message in public chat from this contact.

The systemTags if contains information about blocked, should be preserved

Reply:
> That means that if you remove someone from your friend list (someone you can have 1:1 chat with), you will at the same time remove him from your blocked list, so you will receive message in public chat from this contact.

That is what I asked cause I don't know what is a request, do we need to show messages from blocked contact in a public chat or not?

My logic says that if I block someone he shouldn't be able to send me a message, neither to add me as a contact, but I should be able to see his messages in any public/group chat/channel. That's why I asked what is a request for that?

Reply:
To answer your question:
> That is what I asked cause I don't know what is a request, do we need to show messages from blocked contact in a public chat or not?

I don't think so, at least it seems to be current behaviour. Otherwise the blocked contact feature is kinda useless right?

Note that here I only requested change because of:
```
- contact.systemTags.delete(contact.systemTags.find(contactAdded))
+ contact.systemTags = @[]
```

While it should be:
```
+ contact.systemTags.delete(contact.systemTags.find(contactAdded))
+contact.systemTags.delete(contact.systemTags.find(contactRequest))
```

Reason for that is that if you are contact with someone then decide to block this contact then to remove this contact then the tags: `contactBlocked` needs to stay otherwise the action that you are implemeting is:
Remove this contact from my contact list and unblock him, while the action should only be: Remove this contact from my contact list

Reply:
@alaibe initially I've thought that only my accepted contacts may be blocked. All in all to conclude this discussion, I did a change here. Further, based on new decision made in [issue-3666](https://app.zenhub.com/workspaces/nim-status-client--nim-status-5ea70d29c12ee928df5ae848/issues/status-im/status-desktop/3666) we will update rights for "blocked user".